pkg/keycloak/realm/phaseHandler.go

```diff
@@ -281,7 +289,46 @@ func (ph *phaseHandler) reconcileUserClientRoles(specUser *v1alpha1.KeycloakUser
 	if me.IsNil() {
 		return nil
 	}
-	return me
+	return nil
+}
+
+func (ph *phaseHandler) reconcileUserFederatedIdentities(specUser *v1alpha1.KeycloakUser, realmName string, createOnly bool, authenticatedClient keycloak.KeycloakInterface) error {
+	user, err := authenticatedClient.FindUserByUsername(specUser.UserName, realmName)
+	if err != nil {
+		return err
+	}
+
+	currentFids, err := authenticatedClient.GetUserFederatedIdentities(user.ID, realmName)
+	if err != nil {
+		return err
+	}
+
+	containsFi := func(fidList []v1alpha1.FederatedIdentity, targetFid v1alpha1.FederatedIdentity) bool {
+		for _, fid := range fidList {
+			if reflect.DeepEqual(fid, targetFid) {
+				return true
+			}
+		}
+
+		return false
+	}
+	me := util.NewMultiError()
+	if !createOnly {
+		for _, currentFid := range currentFids {
+			if !containsFi(specUser.FederatedIdentities, currentFid) {
+				me.AddError(authenticatedClient.RemoveFederatedIdentity(currentFid, user.ID, realmName))
+			}
+		}
+	}
+	for _, specFid := range specUser.FederatedIdentities {
+		if !containsFi(currentFids, specFid) {
+			me.AddError(authenticatedClient.CreateFederatedIdentity(specFid, user.ID, realmName))
+		}
+	}
+	if me.IsNil() {
+		return nil
+	}
+	return nil
```

Comment:
Shouldn't we return error at this point?

Reply:
Good spot. Ill update this now :+1: